gips/inventory/dbinv/models.py

```diff
@@ -47,13 +62,57 @@ class Product(models.Model):
     tile    = models.TextField(db_index=True)   # 'h12v04'
     date    = models.DateField(db_index=True)   # of observation, not production
     name    = models.TextField()                # file name including full path
-    status  = models.ForeignKey(Status)         #
-
+    status = models.TextField(validators=[valid_status])
+    
     class Meta:
         # These four columns uniquely identify an asset file
         unique_together = ('driver', 'product', 'tile', 'date')
 
+    # TODO: this is clunky and makes multiple round trips, and is repetitive
+    #       seems to beg to be a trigger
+    def save(self, *args, **kwargs):
+        if self.pk is not None:
+            orig = Product.objects.get(pk=self.pk)
+            super(Asset, self).save(*args, **kwargs)
+            if orig.status != self.status:
+                change = ProductStatusChange(product=self, status=self.status)
+                change.save()
+        else:
+            super(Asset, self).save(*args, **kwargs)
+            change = ProductStatusChange(product=self, status=self.status)
+            change.save()
+
+class AssetDependency(models.Model):
+    """Dependencies of products on specific assets"""
+    product = models.ForeignKey(Product)
+    asset   = models.ForeignKey(Asset)
+    
+    class Meta:
+        unique_together = ('product', 'asset')
+
+        
+class AssetStatusChange(models.Model):
+    """Record of times product status updates"""
```

Comment:
Can you walk me through the use case for this model's uniqueness?  It looks like you can repeat an asset multiple times, but only once per valid status.  Is it expected that no asset can go 'backwards' to a state its already seen?  It seems like going from requested to in-progress to fail then back to in-progress might be a thing, but due to the way the `save` overload works, that last transition would raise an exception.

However, this is probably good enough for our rough and ready prototype.